svm/tests/concurrent_tests.rs

```diff
@@ -97,3 +117,178 @@ fn test_program_cache_with_exhaustive_scheduler() {
+                },
+                AccountMeta {
+                    pubkey: read_account,
+                    is_signer: false,
+                    is_writable: false,
+                },
+                AccountMeta {
+                    pubkey: system_account,
+                    is_signer: false,
+                    is_writable: false,
+                },
+            ],
+            HashMap::from([(sender, Signature::new_unique())]),
+            vec![0],
+        );
+
+        let sanitized_transaction =
+            transaction_builder.build(Hash::default(), (fee_payer, Signature::new_unique()), true);
+        transactions[idx % THREADS].push(sanitized_transaction.unwrap());
+        check_data[idx % THREADS].push(CheckTxData {
+            fee_payer,
+            recipient,
+            sender,
+        });
+    }
+
+    let ths: Vec<_> = (0..THREADS)
+        .map(|idx| {
+            let local_batch = batch_processor.clone();
+            let local_bank = mock_bank.clone();
+            let th_txs = std::mem::take(&mut transactions[idx]);
+            let check_results = vec![
+                Ok(CheckedTransactionDetails {
+                    nonce: None,
+                    lamports_per_signature: 20
+                }) as TransactionCheckResult;
+                TRANSACTIONS_PER_THREAD
+            ];
+            let processing_config = TransactionProcessingConfig {
+                recording_config: ExecutionRecordingConfig {
+                    enable_log_recording: true,
+                    enable_return_data_recording: false,
+                    enable_cpi_recording: false,
+                },
+                ..Default::default()
+            };
+            let check_tx_data = std::mem::take(&mut check_data[idx]);
+
+            thread::spawn(move || {
+                let result = local_batch.load_and_execute_sanitized_transactions(
+                    &*local_bank,
+                    &th_txs,
+                    check_results,
+                    &TransactionProcessingEnvironment::default(),
+                    &processing_config,
+                );
+
+                for (idx, item) in result.processing_results.iter().enumerate() {
+                    assert!(item.was_processed());
+                    let inserted_accounts = &check_tx_data[idx];
+                    for (key, account_data) in &item.as_ref().unwrap().loaded_transaction.accounts {
+                        if *key == inserted_accounts.fee_payer {
+                            assert_eq!(account_data.lamports(), BALANCE - 10000);
+                        } else if *key == inserted_accounts.sender {
+                            assert_eq!(account_data.lamports(), BALANCE - AMOUNT);
+                        } else if *key == inserted_accounts.recipient {
+                            assert_eq!(account_data.lamports(), BALANCE + AMOUNT);
+                        }
+                    }
+                }
```

Comment:
Just another note on benchmarking. If we wanted to use this test for perf testing, we might want to slide these assertions out of the thread pool. Instead we could just store a `results` vector and stash the result of each thread's SVM execution.

Reply:
Ok so you're really just testing that it can run concurrently on the same readonly accounts?

Reply:
Yes, exactly.